### DISCUSSION TITLE
Workflow for working with many pdf documents

### DISCUSSION THREAD

**Discussion by DavidJanz** [i3/i3#5929] [Q&A] [answered]
I'm in the process of checking out i3-wm (on Arch) and seeing if I can move my usual workflow over. I previously used the KDE/Gnome desktop environments, so am more used to working with floating windows. I'm a researcher. I really like how clutter-free and fast it is, and that works well for my writing process. 

However, a large part of my workflow is reading papers from arxiv.org. My usual workflow is that I would download them using Firefox, click on them in the download dialog, which would open the paper in a new (floating) window in evince (or some other pdf view). I would scroll through the paper a bit, then find another one and continue reading, then maybe tab back to the first paper, etc. Over the course of a session, I might open, say, 20 (or 100) papers in that manner (maybe closing the less interesting ones along the way). Alt-tabbing through them isn't ideal, but they're usually in a logical enough order (that in which I came across them). But those desktop environments would have a way to 'zoom out' and see a grid of the opened tabs with previews, so I could find a paper that way if needed. An important issue is that these papers will all have hard-to-distinguish numerical names, e.g., "2311.08376.pdf", so a mechanisms that rely on file-names to select between the documents might be troublesome. 

Does anyone have experience in organising this kind of workflow on i3-wm, and might be able to provide some advice on how to make it work well?

**Comment by waterkip** [comment]
Yesterday I discovered rofi, and it has a function called `rofi -show windows`, it is sorta like alt-tab, but also isn't. Give it a try. It is pure textual, but it allows you to type a filter and you can press up/down on your keyboard to goto a window.

**Reply by DavidJanz** [reply]
Thanks, that looks like a nice way of quickly switching to a tab!

**Comment by okraits** [comment]
If you have that many windows open, I suggest to use the stacked or tabbed layout. An advantage is that you can modify the order of the windows in the layout to your liking.
If you want to group several windows together, you can do that by moving them to a separate workspace.

Additionally, you could use marks to add a more descriptive name to each windows: https://i3wm.org/docs/userguide.html#vim_like_marks

**Reply by DavidJanz** [reply]
Thanks, marks look close to what I need, I'll spend some time playing with them soon. There's one thing I'm trying to achieve that seems to be completely against the "tree" structure of i3, but perhaps it might be possible: I have my screen split into two, the top has a number of pdfs in a tabbed mode, whereas the bottom has ranger open, browsing through my pdfs documents. Now, can I open a pdf document from ranger as a tab in the top container, rather as a branch in the ranger (bottom) container? (my vocabulary might be off, but as a new branch from the other branch of the tree, not the current). That set-up would be close to perfect. Can this be done?

**Reply by okraits** [reply, marked as answer]
Yes, you could use https://i3wm.org/docs/userguide.html#for_window to execute a command which moves the evince window into the position you want it to be. Example:

for_window [class="Evince"] move up

So each time a new evince window shows up, it is moved one step to the top.

**Reply by DavidJanz** [reply]
Thank you, this did the trick! Together with no_focus, and a script that renames downloaded papers to more sensible things, this is working well.

One last minor thing that I don't fully understand is the behaviour of entering tabbed mode. 
Suppose that I: 1. open ranger, 2. open paper A from ranger, 3. open paper B in ranger (with the above command enabled). I now have the way split 3-ways vertically, top to bottom going (A, B, ranger). Now, if I enter tabbed mode ($mod+w), no matter where my focus is, what I get is all of (A, B, ranger) being tabbed, and not just (A, B). How does tabbed-mode decide what the 'scope' of the tabbing is?  Is there a way to tell i3 that these [class="Evince"] windows being opened are to be tabbed, so that I get (Tabbed(A, B), ranger), perhaps even without having to enter tabbed mode manually, in place of Tabbed(A, B,  ranger)?

Currently, of course, the above is easily solved by, after entering tabbed mode, focusing on the ranger tab and moving it down. But I'd be keen to know if there's a nice solution.

**Reply by okraits** [reply]
If you open a new workspace and open two windows (ranger and evince), you have one parent container which has a vertical tiled layout, with two client windows inside. You need to execute `splitv` or `splith` on the evince window to make it a split container which can have a layout on its own (tabbed in your case) and can contain several windows. Then you can execute `layout tabbed` to set the tabbed layout on the split container.

See https://i3wm.org/docs/userguide.html#_splitting_containers and the screencast https://www.youtube.com/watch?v=AWA8Pl57UBY
